sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ProtoDescriptorGenerator.scala

```diff
@@ -108,12 +110,54 @@ object ProtoDescriptorGenerator {
         .build()
     }
 
+    private def buildMethodHttpOptions(component: Class[_], method: Method): MethodOptions = {
+
+      // TODO: also read class level mappings
+      val ruleBuilder = HttpRule.newBuilder()
+
+      Option(AnnotatedElementUtils.findMergedAnnotation(method, classOf[RequestMapping]))
+        .map { mapping =>
+          // TODO: just flatting the mapping for now, will need validate later
+          val fullPath = mapping.path().mkString("/")
+
+          // FIXME: naively picking the first only for now
+          val httpMethod = mapping.method()(0)
+          httpMethod match {
+            case RequestMethod.GET   => ruleBuilder.setGet(fullPath)
+            case RequestMethod.POST  => ruleBuilder.setPost(fullPath)
+            case RequestMethod.PUT   => ruleBuilder.setPut(fullPath)
+            case RequestMethod.PATCH => ruleBuilder.setPatch(fullPath)
+
+            // TODO: to implement when we support ValueEntity / Views deletion
+            case RequestMethod.DELETE =>
+              throw new IllegalArgumentException(s"Unsupported Http method: $httpMethod")
+
+            // those below are not supported in by the proto http annotations
+            case RequestMethod.OPTIONS | RequestMethod.HEAD | RequestMethod.TRACE =>
+              throw new IllegalArgumentException(s"Unsupported Http method: $httpMethod")
+          }
+
+        }
+        .getOrElse {
+          // non-annotated methods get an url like in gRPC, eg: full-qualified-name/method-name
+          ruleBuilder.setPost(s"/${component.getName}/${method.getName}")
```

Comment:
hmm, wondering now why we are even bothering doing that. 
Would we get this already for free from the proto?

Reply:
Well, I think the initial thought was to provide an easier name for the user and not so long as the fully-qualified name. 

But when implementing, I decide to use the FQN to avoid any conflict, but now I realise that it doesn't make sense to add it if we use the FQN.